app/sitemap.py

```diff
@@ -15,142 +15,74 @@
-                    then + timedelta(seconds=offset_seconds))
-                shards.append(shard)
-            self.render('sitemap-index.xml', shards=shards,
-                        static_lastmod=format_sitemaps_datetime(then),
-                        static_map_files=sitemap_info.static_sitemaps)
-        else:
-            shard_index = int(requested_shard_index)
-            assert 0 <= shard_index < 50000    #TODO: nicer error (400 maybe)
-            persons = []
-            time_lower = \
-                then + timedelta(seconds=shard_size_seconds * shard_index)
-            time_upper = time_lower + timedelta(seconds=shard_size_seconds)
-            query = Person.all_in_repo(self.repo
-                         ).filter('last_modified >', time_lower
-                         ).filter('last_modified <=', time_upper
-                         ).order('last_modified')
-            fetched_persons = query.fetch(self._FETCH_LIMIT)
-            while fetched_persons:
-                persons.extend(fetched_persons)
-                last_value = fetched_persons[-1].last_modified
-                query = Person.all_in_repo(self.repo
-                             ).filter('last_modified >', last_value
-                             ).filter('last_modified <=', time_upper
-                             ).order('last_modified')
-                fetched_persons = query.fetch(self._FETCH_LIMIT)
-            urlinfos = [
-                {'person_record_id': p.record_id,
-                 'lastmod': format_sitemaps_datetime(p.last_modified)}
-                for p in persons]
-            self.render('sitemap.xml', urlinfos=urlinfos)
+        langs = const.LANGUAGE_ENDONYMS.keys()
+        urlpaths = []
+        # Include the root/global homepage.
+        urlpaths.append({lang: '?lang=%s' % lang for lang in langs})
+        # Include the repo homepages.
+        for repo in Repo.list_launched():
+            urlpaths.append({
+                lang: '%s?lang=%s' % (repo, lang) for lang in langs})
+        self.render('sitemap.xml', urlpaths=urlpaths)
+
 
 class SiteMapPing(BaseHandler):
-    """Pings the index server with sitemap files that are new since last ping"""
-    _INDEXER_MAP = {'google': 'http://www.google.com/webmasters/tools/ping?',
-                    'not-specified': ''}
+    """Pings the index server."""
+    _INDEXER_MAP = {
+        'bing': 'http://www.bing.com/ping?sitemap=%s',
+        'google': 'http://www.google.com/ping?sitemap=%s',
+    }
+
+    repo_required = False
+
+    @staticmethod
+    def add_ping_tasks():
+        for search_engine in SiteMapPing._INDEXER_MAP:
+            name = 'sitemapping-%s-%s' % (search_engine, int(time.time()*1000))
+            taskqueue.add(name=name, method='GET', url='/global/sitemap/ping',
+                          params={'search_engine': search_engine})
 
     def get(self):
         search_engine = self.request.get('search_engine')
```

Comment:
Nit: maybe add a string constant for this 'search_engine' parameter, to protect against typos?

Reply:
Done.